solidity/contracts/system/VendingMachine.sol

```diff
@@ -203,6 +201,6 @@ contract VendingMachine is TBTCSystemAuthority{
             tbtcToken.approve(_depositAddress, tbtcOwed);
         }
 
-        _d.transferAndRequestRedemption(_outputValueBytes, _redeemerOutputScript, _finalRecipient);
+        _d.transferAndRequestRedemption(_outputValueBytes, _redeemerOutputScript, msg.sender);
```

Comment:
Just a sanity check: our tests are sufficient to ensure this doesn't break anything, right?

Reply:
should be, yes.
This just restricts functionality, and instasnces where the function was called we were passing owner as `_finalRecipient` (so the same functinality as msg.sender). Basically, we should be better than we were before